bifrost/lib/turbolinks/controller.ts

```diff
@@ -115,7 +115,8 @@ export class Controller {
       document.querySelector("#proxied-body") &&
       document.head
         .querySelector("meta[name='turbolinks-no-cache']")
-        ?.getAttribute("content") != "no-cache"
+        ?.getAttribute("content") != "no-cache" &&
+      this.pageContext
```

Comment:
this avoid saving a snapshot altogether if pageContext is not present. This backstop should entirely prevent this class of errors by falling back to rendering the page as if it were a fresh navigation.

This scenario can happen by navigating on first page load, in the brief window between `Turbolinks.start` enabling client side navigation and `onBeforeRenderClient` caching the initial hydrated pageContext. This is a slimmer window than "anytime during React hydration" which was the prior state.